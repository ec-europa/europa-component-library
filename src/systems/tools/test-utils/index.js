const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
