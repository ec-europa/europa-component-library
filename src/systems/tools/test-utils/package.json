{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
