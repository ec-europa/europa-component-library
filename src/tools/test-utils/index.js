const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
