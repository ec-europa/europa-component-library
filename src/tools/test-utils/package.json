{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
