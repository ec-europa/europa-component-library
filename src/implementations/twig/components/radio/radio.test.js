import { merge, renderTwigFileAsNode } from '@ecl/test-utils';

// Import data for tests

import dataDefault from '@ecl/specs-component-radio/demo/data--default';
import dataBinary from '@ecl/specs-component-radio/demo/data--binary';

const dataInvalid = { ...dataDefault, invalid: true };
const dataBinaryInvalid = { ...dataBinary, invalid: true };

describe('Radio', () => {
  const template = '@ecl/radio/radio-group.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataDefault, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataDefault, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Invalid', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataInvalid)).resolves.toMatchSnapshot();
    });
  });

  describe('Binary', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataBinary)).resolves.toMatchSnapshot();
    });
  });

  describe('Binary invalid', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataBinaryInvalid)).resolves.toMatchSnapshot();
    });
  });
});
