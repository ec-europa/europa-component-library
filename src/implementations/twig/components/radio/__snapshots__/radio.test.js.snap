// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Binary invalid renders correctly 1`] = `
<fieldset
  aria-describedby="helper-id-1"
  class="ecl-form-group"
>
  <legend
    class="ecl-form-label ecl-form-label--invalid"
  >
    Do you need help?
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </legend>
  <div
    class="ecl-help-block"
    id="helper-id-1"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-feedback-message"
  >
    Error message for the group
  </div>
  <div
    class="ecl-radio ecl-radio--binary"
  >
    <input
      checked=""
      class="ecl-radio__input"
      id="radio-binary-1"
      name="radio-group-1"
      type="radio"
      value="yes"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-1"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      Yes
    </label>
  </div>
  <div
    class="ecl-radio ecl-radio--binary"
  >
    <input
      class="ecl-radio__input"
      id="radio-binary-2"
      name="radio-group-1"
      type="radio"
      value="no"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-2"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      No
    </label>
  </div>
</fieldset>
`;

exports[`Radio Binary renders correctly 1`] = `
<fieldset
  aria-describedby="helper-id-1"
  class="ecl-form-group"
>
  <legend
    class="ecl-form-label"
  >
    Do you need help?
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </legend>
  <div
    class="ecl-help-block"
    id="helper-id-1"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-radio ecl-radio--binary"
  >
    <input
      checked=""
      class="ecl-radio__input"
      id="radio-binary-1"
      name="radio-group-1"
      type="radio"
      value="yes"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-1"
    >
      <span
        class="ecl-radio__box"
      />
      Yes
    </label>
  </div>
  <div
    class="ecl-radio ecl-radio--binary"
  >
    <input
      class="ecl-radio__input"
      id="radio-binary-2"
      name="radio-group-1"
      type="radio"
      value="no"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-2"
    >
      <span
        class="ecl-radio__box"
      />
      No
    </label>
  </div>
</fieldset>
`;

exports[`Radio Default renders correctly 1`] = `
<fieldset
  aria-describedby="helper-id-1"
  class="ecl-form-group"
>
  <legend
    class="ecl-form-label"
  >
    Select your country
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </legend>
  <div
    class="ecl-help-block"
    id="helper-id-1"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name="radio-group-1"
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box"
      />
      Luxembourg
    </label>
    <div
      class="ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name="radio-group-1"
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box"
      />
      Belgium
    </label>
    <div
      class="ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name="radio-group-1"
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label ecl-radio__label--disabled"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-radio__help ecl-radio__help--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;

exports[`Radio Default renders correctly with extra attributes 1`] = `
<fieldset
  aria-describedby="helper-id-1"
  class="ecl-form-group"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <legend
    class="ecl-form-label"
  >
    Select your country
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </legend>
  <div
    class="ecl-help-block"
    id="helper-id-1"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name="radio-group-1"
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box"
      />
      Luxembourg
    </label>
    <div
      class="ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name="radio-group-1"
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box"
      />
      Belgium
    </label>
    <div
      class="ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name="radio-group-1"
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label ecl-radio__label--disabled"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-radio__help ecl-radio__help--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;

exports[`Radio Default renders correctly with extra class names 1`] = `
<fieldset
  aria-describedby="helper-id-1"
  class="ecl-form-group custom-class custom-class--test"
>
  <legend
    class="ecl-form-label"
  >
    Select your country
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </legend>
  <div
    class="ecl-help-block"
    id="helper-id-1"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name="radio-group-1"
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box"
      />
      Luxembourg
    </label>
    <div
      class="ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name="radio-group-1"
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box"
      />
      Belgium
    </label>
    <div
      class="ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name="radio-group-1"
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label ecl-radio__label--disabled"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-radio__help ecl-radio__help--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;

exports[`Radio Invalid renders correctly 1`] = `
<fieldset
  aria-describedby="helper-id-1"
  class="ecl-form-group"
>
  <legend
    class="ecl-form-label ecl-form-label--invalid"
  >
    Select your country
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </legend>
  <div
    class="ecl-help-block"
    id="helper-id-1"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-feedback-message"
  >
    This is the error message
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name="radio-group-1"
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      Luxembourg
    </label>
    <div
      class="ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name="radio-group-1"
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      Belgium
    </label>
    <div
      class="ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name="radio-group-1"
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label ecl-radio__label--disabled"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled ecl-radio__box--invalid"
      />
      France (disabled)
    </label>
    <div
      class="ecl-radio__help ecl-radio__help--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;
